tools/topology/topology2/platform/intel/sdw-amp-generic.conf

```diff
@@ -130,21 +144,36 @@ IncludeByKey.PASSTHROUGH {
 						in_valid_bit_depth      32
 					}
 				]
-				num_output_audio_formats 3
-				Object.Base.output_audio_format [
-					{
-						out_bit_depth           16
-						out_valid_bit_depth     16
-					}
-					{
-						out_bit_depth           32
-						out_valid_bit_depth     24
+				IncludeByKey.SDW_AMP_FMT_24 {
+				"true"	{
+						num_output_audio_formats 1
+						Object.Base.output_audio_format [
+							{
+								out_bit_depth           32
+								out_valid_bit_depth     24
+								out_sample_type		$SAMPLE_TYPE_MSB_INTEGER
+								out_fmt_cfg		"$[($out_channels | ($out_valid_bit_depth * 256)) | ($out_sample_type * 65536)]"
+							}
+						]
 					}
-					{
-						out_bit_depth           32
-						out_valid_bit_depth     32
+				"false"	{
+						num_output_audio_formats 3
+						Object.Base.output_audio_format [
+							{
+								out_bit_depth           16
+								out_valid_bit_depth     16
+							}
+							{
+								out_bit_depth           32
+								out_valid_bit_depth     24
+							}
```

Comment:
Should the SAMPLE_TYPE_MSG_INTEGER defs be added to this definition as well?

Reply:
> Should the SAMPLE_TYPE_MSG_INTEGER defs be added to this definition as well?

Yes, done.